clash-lib/src/Clash/Core/Subst.hs

```diff
@@ -622,6 +623,18 @@ deShadowAlt ::
   (Pat, Term)
 deShadowAlt is = substAlt "deShadowAlt" (mkSubst is)
 
+-- | Ensure that non of the binders of a let-expression shadow each other, nor
+-- conflict with the in-scope set
+deshadowLetExpr ::
+  HasCallStack =>
+  InScopeSet ->
+  [LetBinding] ->
+  Term ->
+  ([LetBinding],Term)
```

Comment:
While nice, I don't think we should use this style just yet, as it makes it impossible to document functions using it.